lib/ash_ai/actions/prompt.ex

```diff
@@ -223,19 +294,126 @@ defmodule AshAi.Actions.Prompt do
   end
 
   # sobelow_skip ["RCE.EEx"]
-  defp get_prompts(input, opts, context) do
+  defp get_messages(input, opts, context) do
+    template_vars = %{input: input, context: context}
+
     case Keyword.get(opts, :prompt, @prompt_template) do
-      {prompt, user_message} ->
-        prompt = EEx.eval_string(prompt, assigns: [input: input, context: context])
+      # Format 1: String (EEx template)
+      prompt when is_binary(prompt) ->
+        system_prompt = EEx.eval_string(prompt, assigns: [input: input, context: context])
+
+        messages = [
+          LangChain.Message.new_system!(system_prompt),
+          LangChain.Message.new_user!("Perform the action")
+        ]
+
+        {messages, system_prompt, "Perform the action"}
```

Comment:
I think what we should do instead here is refactor this such that what is stored in `data` is the "chain" itself, and we just remove the system_prompt and user_message from data. This simplifies a lot, and removes most of the hacky stuff to extract "legacy" prompts. Its fine to break between releases here as we're still sub 1.0 and things will have to be able to change.

Reply:
I don't see that commit in this PR?

Reply:
Either way, I do see the "legacy" logic still here which we should be able to remove right?

Reply:
https://github.com/ash-project/ash_ai/pull/72/commits/e29435bc47cee7e88e474033e198608fceb25a70

Reply:
@zachdaniel ready for review.